.github/workflows/regression-tests.yml

```diff
@@ -0,0 +1,147 @@
+name: Regression tests
+
+on:
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  prepare-commits:
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.prepare.outputs.matrix }}
+    steps:
+      # Special setup for PRs to extract all commits in PR
+      # with the maximum checkout speed
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ github.head_ref }}
+          fetch-depth: ${{ github.event.pull_request.commits }}
+      - name: Prepare commits
+        id: prepare
+        run: |
+          git fetch --depth=1 origin +refs/heads/${{github.base_ref}}:refs/remotes/origin/${{github.base_ref}}
+          COMMITS_TO_TEST=$(git rev-list origin/${{ github.base_ref }}..${{ github.head_ref }} --pretty='"%H"' --no-commit-header)
+          echo "matrix={ \"commit\": [${COMMITS_TO_TEST//$'\n'/, }] }" | tee -a ${GITHUB_OUTPUT}
+
+  regression-tests:
+    runs-on: [ self-hosted, ci-runner-compiler ]
+    container:
+      image: matterlabs/llvm_runner:ubuntu22-llvm15-latest
+      options: -m 110g
+    needs: prepare-commits
+    strategy:
+      fail-fast: false
+      max-parallel: 3
+      matrix: ${{ fromJson(needs.prepare-commits.outputs.matrix) }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ matrix.commit }}
+          fetch-depth: ${{ github.event.pull_request.commits }}
+          path: "llvm"
+
+      - name: Check formatting
+        working-directory: llvm
+        shell: bash
+        run: |
+          git fetch --depth=1 origin +refs/heads/${{github.base_ref}}:refs/remotes/origin/${{github.base_ref}}
+          CHANGED_FILES="$(git diff --name-only origin/${{ github.base_ref }})"
+          CHANGED_FILES="${CHANGED_FILES//$'\n'/,}"
+          echo "Formatting files:"
+          echo "${CHANGED_FILES}"
+          python -m pip install -r ./llvm/utils/git/requirements_formatting_era_llvm.txt
```

Comment:
Agree, it could be very dangerous to use `python` even though I consciously used it in this line because I explicitly set `python` [here](https://github.com/matter-labs/compiler-infra/blob/main/images/llvm_runner/ubuntu22-llvm15.Dockerfile#L63-L67), it's still better to use `python3`. Updated the code.